packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -93,37 +72,77 @@ function AutoInsertingBlocksControl( props ) {
 						);
 
 						if ( autoInsertedBlock ) {
-							clientIds[ block.name ] =
-								autoInsertedBlock.clientId;
+							clientIds = {
+								...clientIds,
+								[ block.name ]: autoInsertedBlock.clientId,
+							};
+						} else if ( getGlobalBlockCount( block.name ) > 0 ) {
```

Comment:
I had this after-thought that we could put:

```js
if ( getGlobalBlockCount( block.name ) === 0 ) {
    return clientIds;
}
```

at the top of the function. If there is no block present in the editor, then we are sure we can show the toggle. In this place, we could simplify the check to:

```js
} else {
```

With proper inline comments that might also simplify following all the edge cases.

Reply:
Makes sense! Applied in 35ebe4e8bc 😊